test/unittests/studyentities/CandID_Test.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * This file contains unit test for the CandID value object.
+ *
+ * PHP Version 7
+ *
+ * @category Tests
+ * @package  StudyEntities
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\StudyEntitites\Candidate\Test;
+
+require_once __DIR__ . '/../../../vendor/autoload.php';
+
+use \PHPUnit\Framework\TestCase;
+use \LORIS\StudyEntities\Candidate\CandID;
+
+/**
+ * Unit test class for the CandID value object
+ *
+ * PHP Version 7
+ *
+ * @category Tests
+ * @package  StudyEntities
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class CandID_Test extends TestCase
+{
+    /*
+     * dataProviders for constructor invalid values
+     */
+    public function invalidValues(): array
+    {
+        return array(
+                ['A'],
+                [1],
+                [000000],
```

Comment:
I'm a little bit confused about the test cases. For example this case is invalid, but `'000000'` (string value of the same ID) is valid. This makes me think that CandIDs should be created via string representations of numbers, but below you use `123456` (not `'123456'`) to create a CandID.

I'm not sure what implications this has for CandIDs with leading 0s. PHP gets really weird about stuff like this, e.g. 

```php
'000000' == 0;
```
will return true
```php
'0123456' == 123456;
```
returns true, and
```php
'0123456' == 0123456
```
return false.

To avoid possible type issues I think it would be good to be really explicit about the types that are allowed here. Another thing we could do is enforce that a Candidate ID is at least `100000` which is the current method used in the `Candidate` class. Either way it should be documented. I'm sure similar issues will come up with PSCID/ExternalID.

Reply:
Otherwise this looks very good!

Reply:
@johnsaigle I don't mind adding a rule for `>= 100000`.
Regarding the test that calls the constructor with `123456`, the value is cast as a string in the function so there is no way, to my knowledge, to check that. I did some tests using 012345 and the value becomes the string "5349" which is the decimal value of octal 12345. It fails the preg_match but calling it with 0777777 would pass. Maybe Phan would cast that. do you know?

Reply:
I think treating CandID as always a string instead of an int is the way to go. We never actually do math with it so it doesn't need to be an int. The string representation won't remove leading `0`s or get cast to different numerical representations.